src/cdk-experimental/listbox/listbox.ts

```diff
@@ -0,0 +1,152 @@
+    '(keydown)': 'state.onKeydown($event)',
+    '(mousedown)': 'state.onMousedown($event)',
+  },
+})
+export class CdkListbox implements ListboxInputs, OnDestroy {
+  /** The directionality (LTR / RTL) context for the application (or a subtree of it). */
+  private _dir = inject(Directionality);
+
+  /** Whether the list is vertically or horizontally oriented. */
+  orientation = input<'vertical' | 'horizontal'>('vertical');
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable = input<boolean>(false);
+
+  /** Whether focus should wrap when navigating. */
+  wrap = input<boolean>(true);
+
+  /** Whether disabled items in the list should be skipped when navigating. */
+  skipDisabled = input<boolean>(true);
+
+  /** The focus strategy used by the list. */
+  focusMode = input<'roving' | 'activedescendant'>('roving');
+
+  /** The selection strategy used by the list. */
+  selectionMode = input<'follow' | 'explicit'>('follow');
+
+  /** The amount of time before the typeahead search is reset. */
+  delay = input<number>(0.5); // Picked arbitrarily.
+
+  /** The ids of the current selected items. */
+  selectedIds = model<string[]>([]);
+
+  /** The current index that has been navigated to. */
+  activeIndex = model<number>(0);
+
+  /** The CdkOptions nested inside of the CdkListbox. */
+  private _cdkOptions = contentChildren(CdkOption, {descendants: true});
+
+  /** The Option UIPatterns of the child CdkOptions. */
+  items = computed(() => this._cdkOptions().map(option => option.state));
+
+  /** A signal wrapper for directionality. */
+  directionality = signal<'ltr' | 'rtl'>('ltr');
+
+  /** Emits when the list has been destroyed. */
+  private readonly _destroyed = new Subject<void>();
+
+  /** Whether the listbox is disabled. */
+  disabled = input<boolean>(false);
+
+  /** The Listbox UIPattern. */
+  state: ListboxPattern = new ListboxPattern(this);
+
+  constructor() {
+    this._dir.change
+      .pipe(startWith(this._dir.value), takeUntil(this._destroyed))
+      .subscribe(value => this.directionality.set(value));
+  }
+
+  ngOnDestroy() {
+    this._destroyed.complete();
+  }
+}
+
+/** A selectable option in a CdkListbox. */
+@Directive({
+  selector: '[cdkOption]',
+  exportAs: 'cdkOption',
+  host: {
+    'role': 'option',
+    'class': 'cdk-option',
+    '[attr.aria-selected]': 'state.selected()',
+    '[attr.tabindex]': 'state.tabindex()',
+    '[attr.aria-disabled]': 'state.disabled()',
+  },
+})
+export class CdkOption {
+  /** A reference to the option element. */
+  private _elementRef = inject(ElementRef);
+
+  /** The parent CdkListbox. */
+  private _cdkListbox = inject(CdkListbox);
+
+  /** Whether an item is disabled. */
+  disabled = input<boolean>(false);
+
+  /** The text used by the typeahead search. */
+  label = input<string>();
+
+  /** The text used by the typeahead search. */
+  searchTerm = computed(() => this.label() ?? this.element().textContent);
```

Comment:
Should we make the label required so we don't have to hit the DOM every time the `computed` runs?

Reply:
IMO the devex is worth the performance cost in this case. If I understand correctly, this computed will only rerun if the label is set/unset or if the element changes (which I would assume is rare)

Reply:
I think that initially it runs so the runtime can figure out the dependencies and afterwards it only runs on changes. That being said, it might be a bit misleading because it's not going to update if the `textContent` changes. I expect it might come up, given that we had to account for it in `mat-option`: https://github.com/angular/components/blob/main/src/material/core/option/option.ts#L267

My thinking was that it's easier to introduce it later than having consumers depend on it and us having to change it later.

Reply:
Another option can be to only derive the label from the input and if there isn't one, we just don't do typeahead.